Sources/NIOHTTPClient/HTTPHandler.swift

```diff
@@ -19,24 +19,40 @@ import NIOHTTP1
 import NIOSSL
 
 public extension HTTPClient {
-    enum Body: Equatable {
-        case byteBuffer(ByteBuffer)
-        case data(Data)
-        case string(String)
-
-        var length: Int {
-            switch self {
-            case .byteBuffer(let buffer):
-                return buffer.readableBytes
-            case .data(let data):
-                return data.count
-            case .string(let string):
-                return string.utf8.count
+    struct Body {
```

Comment:
Nit: this is really a personal preference (although I think NIO style might mandate this) but I much prefer having explicit access modifiers, and just `extension` instead of `public extension`. I'm not clever enough to remember Swift's access control rules without having to think twice every time...

Reply:
good idea, fixed, thanks!

Reply:
yes please, I think there should never be any word in front of `extension`. `public extension` for example is really bad because it means that you can add public functions without the word `public` appearing in a diff and that's an issue for all SemVer stuff.